swap/peer.go

```diff
@@ -116,12 +117,12 @@ func (p *Peer) setPendingCheque(cheque *Cheque) error {
 
 // getLastSentCumulativePayout returns the cumulative payout of the last sent cheque or 0 if there is none
 // the caller is expected to hold p.lock
-func (p *Peer) getLastSentCumulativePayout() uint64 {
+func (p *Peer) getLastSentCumulativePayout() *uint256.Uint256 {
 	lastCheque := p.getLastSentCheque()
 	if lastCheque != nil {
 		return lastCheque.CumulativePayout
 	}
-	return 0
+	return uint256.New()
```

Comment:
There is a semantical difference in return 0 or a nil value here. I personally would rather see a nil pointer, because it explicitly expresses that no last sent cheque was there - however, seeing that 0 really has no implication and that we are not actually taking advantage of the semantical difference of the nil value in any way I see, I think it is fine to leave it as is.

Reply:
indeed, i also thought about this. and as a general rule, i always prefer to opt for a `nil` return. 

however, i noticed 2 things in this case:
- `getLastSentCumulativePayout` does not return an `error`. this means that we can't really tell if `nil` means something went wrong, or if the result is correct and (just) empty. this wouldn't be the case with a result of a `(pointer, error)` tuple.
- even in case of `nil` `lastCheque`, the result for `getLastSentCumulativePayout` is added to the `price` variable in the `createCheque` function. the problem here is that calling  `Add` with a `nil` value will cause a crash (this is true for `big.Int` as well).